### PR TITLE
Implemented error handling for classical t-tests

### PR DESCRIPTION
**Pull request by koenderks**
error handling implementation for one sample t-test, paired samples t-test & independent samples t-test.

### CODE REVIEW

**Comment by TimKDJ** [comment]
@koenderks there are 3 bayesian files in your PR, although you state that this is only classical.
Is this an error (I think it is given there are >5500 changes in 2 of the Bayesian files)?
Can you remove the files you do not wish to merge, so I know what to really look at?
Cheers!

File: JASP-Engine/JASP/R/ttestonesample.R [hunk 1]
```diff
@@ -487,8 +482,19 @@ TTestOneSample <- function(dataset = NULL, options, perform = "run",
 	for (i in .indices(options$variables)) {
 
 		var <- options$variables[[i]]
-
 		descriptivesPlot <- list("title" = var)
+		#### new error handling
+		errors <- .hasErrors(dataset, perform, message = 'short', type = c('observations', 'variance', 'infinity'),
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/ttestonesample.R]: From what I can see the plot below this would still get made even when errors are found.
Is that intended?

**Comment by TimKDJ** [comment]
I can't see what you changed [in your latest commit](https://github.com/jasp-stats/jasp-desktop/pull/1822/files/993793386da270b86c7b5ede378a55d866df4310..2f84eb56886ecd0b19ee9ad1ad19fcbfd962d78a)... It shows your entire files as changed again.

File: JASP-Engine/JASP/R/ttestindependentsamples.R [hunk 2]
```diff
@@ -623,18 +623,16 @@ TTestIndependentSamples <- function(dataset = NULL, options, perform = "run",
...
-					row.footnotes <- list(W = list(foot.index), p = list(foot.index))
-					error <- TRUE
-
-				} else if (length(data) > 5000) {
-					err <- .generateErrorMessage(type='levene', observations.amount='> 5000', variables=variable, grouping=factor)
-					foot.index <- .addFootnote(footnotes, err)
-					row.footnotes <- list(W = list(foot.index), p = list(foot.index))
-					error <- TRUE
+				
+				errors <- .hasErrors(dataset, perform, message = 'short', type = c('observations', 'variance', 'infinity'),
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/ttestindependentsamples.R]: Also, in your .hasErrors() calls in the independent samples t-test you do not specify the grouping variable.
One group might have 20 observations in one group and in the other zero.

File: JASP-Engine/JASP/R/ttestonesample.R [hunk 3]
```diff
@@ -260,27 +270,16 @@ TTestOneSample <- function(dataset = NULL, options, perform = "run",
 					}
 					res
 				})
+				
+				if (class(row) == "try-error" && is.null(errorMessage)) {
```
Comment (TimKDJ) [inline JASP-Engine/JASP/R/ttestonesample.R]: Also for the paired samples t-test please

File: JASP-Engine/JASP/R/ttestindependentsamples.R (hunk 2 above)
Comment (koenderks) [inline JASP-Engine/JASP/R/ttestindependentsamples.R]: What is the grouping variable in paired and one sample t-test?

Comment (koenderks) [inline JASP-Engine/JASP/R/ttestindependentsamples.R]: just one sample

Comment (TimKDJ) [inline JASP-Engine/JASP/R/ttestindependentsamples.R]: There is none. I worded it poorly; the <3 and >5000 restrictions should be honored in those tests.

**Comment by TimKDJ** [comment]
One more comment and we're good. The last of the three remarks I made previously:

> Also for the paired samples t-test please

File: JASP-Engine/JASP/R/ttestindependentsamples.R [hunk 4]
```diff
@@ -699,13 +699,27 @@ TTestIndependentSamples <- function(dataset = NULL, options, perform = "run",
...
 		}
-
+		
 		for (var in .indices(variables)) {
-			descriptivesPlot <- list("title" = variables[var])
-			descriptivesPlot[["width"]] <- options$plotWidth
-			descriptivesPlot[["height"]] <- options$plotHeight
-			descriptivesPlot[["custom"]] <- list(width = "plotWidth", height = "plotHeight")
-
+
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestindependentsamples.R]: Please change var to variable or variableName, there's also a var function in stats::var

File: JASP-Engine/JASP/R/ttestpairedsamples.R [hunk 5]
```diff
@@ -192,6 +192,11 @@ TTestPairedSamples <- function(dataset = NULL, options, perform = "run",
 
 		p1 <- pair[[1]]
 		p2 <- pair[[2]]
+		
+		errors <- .hasErrors(dataset, perform, message = 'short', type = c('observations', 'variance', 'infinity'),
+							 all.target = c(p1,p2),
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestpairedsamples.R]: space after comma

File: JASP-Engine/JASP/R/ttestpairedsamples.R [hunk 6]
```diff
@@ -239,7 +253,7 @@ TTestPairedSamples <- function(dataset = NULL, options, perform = "run",
 					})
 
 					## if testing raised an error, we extract information from it
-					if (class(r) != "try-error" && is.na(r$statistic) == FALSE) {
+					if (class(r) != "try-error") {
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestpairedsamples.R]: if (isTryError(r)) {

By the way I don't like objects names referred to as "r". Perhaps call it fit or result

File: JASP-Engine/JASP/R/ttestpairedsamples.R [hunk 7]
```diff
@@ -272,32 +286,15 @@ TTestPairedSamples <- function(dataset = NULL, options, perform = "run",
...
-						} else if (errorMessage == "data are essentially constant") {
 
-							err <- paste0("t-statistic is undefined - one or both of ",
-										  "the variables contain all the same value (zero variance)")
-
-						} else if (errorMessage == "not enough 'x' observations") {
-
-							err <- paste0("t-statistic is undefined - one or both of ",
-										  "the variables contain only one value")
+					} else if (class(r) == "try-error") {
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestpairedsamples.R]: isTryError(r) it also looks at inheritence

File: JASP-Engine/JASP/R/ttestpairedsamples.R [hunk 8]
```diff
@@ -516,7 +509,19 @@ TTestPairedSamples <- function(dataset = NULL, options, perform = "run",
 
 			c1 <- dataset[[ .v(pair[[1]]) ]]
 			c2 <- dataset[[ .v(pair[[2]]) ]]
+			###
+			errors <- .hasErrors(dataset, perform, message = 'short', type = c('observations', 'variance', 'infinity'),
+								 all.target = c(pair[[1]],pair[[2]]),
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ttestpairedsamples.R]: comma space

**Pull request merged**